ethereum/core/src/main/java/org/hyperledger/besu/ethereum/core/Transaction.java

```diff
@@ -485,10 +485,16 @@ public BigInteger getV() {
     }
 
     final BigInteger recId = BigInteger.valueOf(signature.getRecId());
-    if (chainId.isEmpty()) {
-      return recId.add(REPLAY_UNPROTECTED_V_BASE);
+
+    if (transactionType != null && transactionType != TransactionType.FRONTIER) {
+      // EIP-2718 typed transaction, return yParity:
```

Comment:
Is it really transaction envelope or EIP1559 type transactions ?

Reply:
It is 2930 transactions also, that was the failing reference test that highlighted this